server/fly.io/src/main.rs

```diff
@@ -0,0 +1,79 @@
+use std::collections::BTreeMap;
+use std::net::SocketAddr;
+
+use axum::{
+    extract::Query,
+    routing::{get, post},
+    Router,
+};
+use influxrs::{InfluxClient, Measurement};
+
+fn main() {
+    let rt = tokio::runtime::Runtime::new().unwrap();
+    let task = rt.spawn(async move {
+        let app = Router::new()
+            .route("/", get(root))
+            .route("/record-install", get(record_install))
+            .route("/record-install", post(record_install));
+
+        // ipv6 + ipv6 any addr
+        let addr = SocketAddr::from(([0, 0, 0, 0, 0, 0, 0, 0], 8080));
+        axum::Server::bind(&addr)
+            .serve(app.into_make_service())
+            .await
+            .unwrap();
+    });
+    rt.block_on(task).unwrap();
+}
+
+async fn root() -> &'static str {
+    "Hello, World!"
+}
+
+fn get_env(key: &str) -> String {
+    std::env::var(key).expect(&format!("{key} must be set"))
+}
+
+async fn record_install(Query(params): Query<BTreeMap<String, String>>) -> String {
+    println!("Hi there {params:?}");
```

Comment:
I think we could use tracing + + tracing-subscriber + tracing-appender for logging.

tracing is like the crate log, while tracing-appender provides non-blocking writer implementation (to prevent blocking the web server), and tracing-subscriber provides the logging implementation.

tracing-subscriber could also print out json as well to make it machine readable.

Reply:
I think I would prefer human readable logs for this server for now.

I don't think it's ever going to be high volume enough to warrant non-blocking logging: I think that the time for cargo-binstall to start up and make an https connection to the log server will dwarf any time saved by async logging.

I would accept patches to switch to `tracing` though, as long as it doesn't make debugging with the fly.io CLI/web tools harder. I think you still have an outstanding invitation to the fly.io org according to https://fly.io/dashboard/cargo-quickinstall/team. I can try re-sending it if you want.

Reply:
> I would accept patches to switch to tracing though, as long as it doesn't make debugging with the fly.io CLI/web tools harder.

👍 I will leave this open so that I don't forget that, though I do think having non-blocking logging is beneficial:

Writing to stdout/stderr is often slower than expected.

> I think you still have an outstanding invitation to the fly.io org according to https://fly.io/dashboard/cargo-quickinstall/team. I can try re-sending it if you want.

Sorry somehow I didn't get it.
Is the invitation linked with my GitHub account, or to my email Jiahao_XU@outlook.com?

Can you send it again please?

Reply:
The original request was sent to Jiahao_XU@outlook.com . I revoked it and sent it again. Maybe it went to spam?

Reply:
Thanks, I've received it now.

I looked it up in my spam but somehow didn't find it (maybe missed it?

Reply:
backlogged logging as https://github.com/cargo-bins/cargo-quickinstall/issues/278